src/react/useSpace.ts

```diff
@@ -0,0 +1,48 @@
+import { useContext, useEffect, useRef } from 'react';
+import { SpaceContext } from './contexts/SpaceContext.js';
+import { useChannelState } from './useChannelState.js';
+import { useConnectionState } from './useConnectionState.js';
+
+import type { ErrorInfo } from 'ably';
+import type { Space } from '..';
+import type { UseSpaceCallback, UseSpaceOptions } from './types.js';
+
+interface UseSpaceResult {
+  space?: Space;
+  enter?: Space['enter'];
+  leave?: Space['leave'];
+  updateProfileData?: Space['updateProfileData'];
+  connectionError: ErrorInfo | null;
+  channelError: ErrorInfo | null;
+}
+
+export const useSpace = (callback?: UseSpaceCallback, options?: UseSpaceOptions): UseSpaceResult => {
```

Comment:
What would the error msg look like if you used this hook without a `SpaceProvider`? I haven't tested it myself but IIRC the error message is usually a bit non-obvious so I would recommend handling this case explicitly.